### PR TITLE
Cleanup and unify ArgReader and ArgWriter interfaces

### PR DESCRIPTION
**Pull request by jcorbin**
- use ArgReader instead of io.ReadCloser
- define both in the arguments.go file

r @prashantv

### CODE REVIEW

**Comment by jcorbin** [comment]
Towards #173

File: arguments.go [hunk 1]
```diff
@@ -27,15 +27,27 @@ import (
 	"io/ioutil"
 )
 
+type ArgReader interface {
```
Comment (prashantv) [inline arguments.go]: Can you add comments to the exported interfaces here and below. (`go vet` / `golint` should complain about this too, suggest setting those up to run on save)

**Comment by prashantv** [comment]
cc @akshayjshah 

Can you also update the changelog,
https://github.com/uber/tchannel-go/blob/master/CHANGELOG.md

**Comment by jcorbin** [comment]
Added comments

File: arguments.go [hunk 2]
```diff
@@ -27,15 +27,31 @@ import (
...
+
+// ArgWriter is the interface for the arg2 and arg3 streams on an OutboundCall
+// and an InboundCallResponse
+type ArgWriter interface {
+	io.WriteCloser
+
+	// Flush flushes the currently written bytes without waiting for the frame
+	// to be filled.
+	Flush() error
+}
```
Comment (akshayjshah) [inline arguments.go]: :+1: for keeping these in the same place.

File: arguments.go [hunk 3]
```diff
@@ -27,15 +27,31 @@ import (
 	"io/ioutil"
 )
 
+// ArgReader is the interface for the arg2 and arg3 streams on an
+// OutboundCallResponse and an InboundCall
+type ArgReader interface {
+	io.ReadCloser
+}
```
Comment (akshayjshah) [inline arguments.go]: Optionally terser as:

```
type ArgReader io.ReadCloser
```

If you're planning to add methods to this interface in a follow-up PR, though, definitely makes sense to keep it as-is.

**Comment by akshayjshah** [comment]
lgtm

File: arguments.go (hunk 3 above)
Comment (prashantv) [inline arguments.go]: +1, we have no plans right now to add more methods, so this seems like a good idea.

**Comment by prashantv** [comment]
lgtm

**Comment by jcorbin** [comment]
Simplified the `ArgReader` type

**Pull request merged**